src/rpc/service_eth.go

```diff
@@ -261,3 +283,51 @@ var DefaultProcessTransaction = func(tx *txtypes.Transaction, ctx context.Contex
 	}
 	return submitTx, nil
 }
+
+func (s *EthService) WaitTillMined(ctx context.Context, tx *types.Transaction, delayInSeconds int) {
+	key, err := s.Cache.Key(tx)
+	if err != nil {
+		utils.Logger.Debug().Msgf("WaitTillMined | error in generating key | err: %v", err)
+		return
+	}
+	value := s.Cache.InclusionCache[key]
+
+	if value == nil {
+		queryTicker := time.NewTicker(time.Duration(delayInSeconds) * 2 * time.Second)
```

Comment:
It would be better to separate this ticker from the delay in seconds, as they have different semantics, and we might want to test them with independent values, would also rename this to WaitMinedInterval for better readability.